packages/headless/src/controllers/commerce/context/cart/headless-cart.ts

```diff
@@ -0,0 +1,151 @@
+import {ArrayValue, NumberValue, RecordValue, Schema } from "@coveo/bueno";
+import {validateOptions} from '../../../../utils/validate-payload';
+import {CommerceEngine} from '../../../../app/commerce-engine/commerce-engine';
+import {buildController, Controller} from '../../../controller/headless-controller';
+import {loadReducerError} from '../../../../utils/errors';
+import {cartReducer as cart} from '../../../../features/commerce/context/cart/cart-slice';
+import {
+  addCartItem,
+  removeCartItem,
+  setCart,
+  updateCartItemQuantity
+} from '../../../../features/commerce/context/cart/cart-actions';
+
+const optionsSchema = new Schema({
+  cart: new ArrayValue({
+    each: new RecordValue({
+      values: {
+        product: new RecordValue({
+          options: {required: true}
+        }),
+        quantity: new NumberValue({
+          required: true,
+          min: 1,
+        })
+      }
+    }),
+  }),
+});
+
+export interface CartOptions {
+  cart?: CartItem[];
+}
+
+export interface CartItem {
+  product: Product;
+  quantity: number;
+}
+
+export interface Product {
+  groupId?: string;
+  productId?: string;
+  sku?: string;
+}
+
+export interface CartProps {
+  /**
+   * The initial options that should be applied to this `Cart` controller.
+   */
+  options?: CartOptions;
+}
+
+/**
+ * The `Cart` controller allows the end user to configure cart data.
+ */
+export interface Cart extends Controller {
+
+  /**
+   * Sets the cart.
+   * @param cart - The new cart.
+   */
+  setCart(cart: CartItem[]): void;
+
+  /**
+   * Adds a cart item.
+   * @param item - The new cart item.
+   */
+  addCartItem(item: CartItem): void;
+
+  /**
+   * Removes a cart item.
+   * @param item - The cart item to remove
+   */
+  removeCartItem(item: CartItem): void;
+
+  /**
+   * Updates a cart item's quantity.
+   * @param item - The cart item to update.
+   */
+  updateCartItemQuantity(item: CartItem): void;
```

Comment:
That's a kinda weird function signature for what it needs to do.

I would see that more as an `updateCartItem` ?

Nothing specific about quantity.